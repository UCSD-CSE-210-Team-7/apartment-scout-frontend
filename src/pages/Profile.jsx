// Import style files and images
import Loading from "../components/Loading";
import userImg from "../img/user.png";
import "../styles/profile-styles.scss";
// Import required dependencies
import { useQuery, useMutation, gql } from "@apollo/client";
import React, { useState } from "react";

// Define GraphQL queries for fetching user details
const QUERY_USER_DETAILS = gql`
  query Me {
    me {
      email
      name
      created_on
      last_login
      is_scout
      calendly_link
      tours {
        tour_id
        tour_review_text
        tour_review_stars
      }
      regions
    }
  }
`;

// Define GraphQL mutations for updating user details
const UPDATE_USER_MUTATION = gql`
  mutation UpdateUser(
    $email: String!
    $name: String
    $is_scout: Boolean
    $regions: [Int]
    $calendly_link: String
  ) {
    updateUser(
      email: $email
      name: $name
      is_scout: $is_scout
      regions: $regions
      calendly_link: $calendly_link
    ) {
      email
      name
      created_on
      last_login
      is_scout
      calendly_link
      tours {
        tour_id
        tour_review_text
        tour_review_stars
      }
      regions
    }
  }
`;

/**
 * Badge component which indicates whether the user is a scout or not
 * @param {string} name - Name to be displayed inside the badge
 * @param {string} state - Optional boolean string which indicates whether the user is scout or not
 * @param {string} color - Optional background color of badge if state is true
 * 
 * @returns {JSX.Element} The JSX element for the Badge component.
 */
function Badge({ name, state = "false", color = "green" }) {
  return (
    <div
      style={{
        padding: "0.4em 0em",
        width: "10em",
        background: state ? color : "grey",
        border: "none",
        margin: "2em 0 0",
        color: "white",
        fontSize: "2em",
        textAlign: "center",
      }}
    >
      {name}
    </div>
  );
}

/**
 * Component that displays a list of zipcodes and provides functionality to add, edit, and delete them.
 *
 * @param {Array<number>} zipcodes - An array of zipcodes to be displayed.
 * @param {function} onChange - A callback function to handle changes to a specific zipcode.
 * @param {function} onDelete - A callback function to handle deletion of zipcode.
 * @param {function} onAdd - A callback function to handle addition of a new zipcode.
 *
 * @returns {JSX.Element} The JSX element for the Zipcodes component.
 */
function Zipcodes({ zipcodes, onChange, onDelete, onAdd }) {
  return (
    <div style={{ padding: "1em", display: "flex", flexDirection: "column" }}>
      <p style={{ margin: 0 }}>regions</p>
      {zipcodes.map((elem, idx) => (
        <div style={{ padding: "0.5em" }}>
          <input
            type="number"
            value={elem}
            onChange={(e) => onChange(idx, parseInt(e.target.value))}
            style={{
              width: "fit-content",
              textAlign: "center",
              border: 0,
              borderBottom: "1px solid black",
            }}
          />
          <button
            onClick={() => onDelete(idx)}
            style={{
              color: "white",
              background: "red",
              borderRadius: "0.5em",
              margin: "0em 0.5em",
              padding: "0em 0.5em",
            }}
          >
            X
          </button>
        </div>
      ))}
      <div style={{ padding: "0.5em" }}>
        <input
          type="number"
          placeholder="add new region"
          value={""}
          onChange={(e) => onAdd(parseInt(e.target.value))}
          style={{
            width: "fit-content",
            textAlign: "center",
            border: 0,
            borderBottom: "1px solid black",
          }}
        />
      </div>
    </div>
  );
}

/**
 * Component that displays an input field with a label, and value
 *
 * @param {string} name - The label for the input field.
 * @param {string} value - The current value of the input field.
 * @param {string} placeholder - The placeholder text for the input box.
 * @param {function} handleChange - A callback function to handle changes to the input value.
 * @param {boolean} disabled - A flag to indicate whether the input box should be disabled.
 * @param {Object} style - An object containing inline styles to be applied to the field.
 *
 * @returns {JSX.Element} The JSX element for the Field component.
 */
function Field({ name, value, placeholder, handleChange, disabled, style }) {
  return (
    <div style={{ padding: "1em" }}>
      <p style={{ margin: 0 }}>{name}</p>
      <input
        type="text"
        readOnly={disabled}
        disabled={disabled}
        placeholder={placeholder}
        value={value}
        onChange={handleChange}
        style={{ ...style }}
      />
    </div>
  );
}

/**
 * Profile component while displays the user details and allows to edit the details
 * @returns {JSX.Element} The JSX element for the Field component.
 */
const Profile = () => {
  const fileInput = React.createRef();

  // Set initial user state
  const [user, setUser] = useState({
    email: "",
    name: "",
    created_on: "",
    last_login: "",
    is_scout: "",
    calendly_link: "",
    tours: [],
    regions: [],
    image: userImg,
  });

  // Fetch the existing user details 
  const { loading: userLoading } = useQuery(QUERY_USER_DETAILS, {
    onCompleted: (data) => {
      setUser({
        ...data.me,
        image: userImg,
      });
    },
  });

  // Use mutation to update user data
  const [updateUserMutation, { loading: updateLoading }] = useMutation(
    UPDATE_USER_MUTATION,
    { onCompleted: () => alert("user updated!") }
  );

  // Function to handle name changes and update user state
  const handleNameChange = (e) => {
    setUser({ ...user, name: e.target.value });
  };

  /*
  const handleImageUpload = (e) => {
    setUser({ ...user, image: e.target.files[0] });
  };
  */

  if (userLoading) return <Loading />;

  return (
    <div
      style={{
        display: "flex",
      }}
    >
      <div
        style={{
          display: "flex",
          flexDirection: "column",
          padding: "0 6em",
          alignItems: "center",
        }}
      >
        <h1
          style={{
            fontSize: "3rem",
            margin: "1em 0em",
            padding: "1em 3em",
            color: "#023E8A",
            background: "#ADE8F4",
          }}
        >
          My Profile
        </h1>
        <img
          src={user.image}
          alt="user"
          style={{
            margin: "2em 0",
            height: "20em",
            width: "20em",
          }}
        />
        <label>
          <input
            type="file"
            style={{
              display: "none",
            }}
            ref={fileInput}
          />
          <h1
            style={{
              color: "white",
              cursor: "pointer",
            }}
          >
            Edit profile photo
          </h1>
        </label>
        <Badge name="Scout" state={user.is_scout} color="green" />
      </div>

      <div
        style={{
          border: "1px solid black",
          display: "flex",
          flexDirection: "column",
          flexGrow: 1,
          padding: "2em",
          background: "white",
          fontSize: "2em",
        }}
      >
        <h1>Personal Info</h1>
        <Field name="name" value={user.name} handleChange={handleNameChange} />
        <Field disabled name="email" value={user.email} />
        <Field
          name="Calendly link"
          placeholder="Paste Calendly link here"
          value={user.calendly_link}
          handleChange={(e) =>
            setUser({ ...user, calendly_link: e.target.value })
          }
          style={{ width: "100%" }}
        />
        <Zipcodes
          zipcodes={user.regions}
          onAdd={(val) => {
            setUser({
              ...user,
              regions: [...user.regions, val],
            });
          }}
          onDelete={(i) => {
            setUser({
              ...user,
              regions: [
                ...user.regions.slice(0, i),
                ...user.regions.slice(i + 1),
              ],
            });
          }}
          onChange={(i, val) => {
            setUser({
              ...user,
              regions: [
                ...user.regions.slice(0, i),
                val,
                ...user.regions.slice(i + 1),
              ],
            });
          }}
        />
        <button
          style={{
            alignSelf: "center",
            padding: "0.5em 1em",
            background: "#023E8A",
            color: "white",
            borderRadius: "1em",
          }}
          onClick={() => {
            updateUserMutation({
              variables: user,
            });
          }}
          disabled={updateLoading}
        >
          {updateLoading ? "Saving..." : "Save"}
        </button>
      </div>
    </div>
  );
};

export default Profile;
