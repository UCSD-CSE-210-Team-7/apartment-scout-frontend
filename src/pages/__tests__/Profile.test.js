import { render } from "../../test_setup";
import {
  render as _r,
  screen,
  act,
  fireEvent,
  waitForElementToBeRemoved,
  waitFor,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import {
  MemoryRouter,
  BrowserRouter,
  RouterProvider,
  createMemoryRouter,
  createRoutesFromElements,
  Routes,
  Route,
} from "react-router-dom";

import { data } from "../../dummy_data/users.json";
import Profile, { QUERY_USER_DETAILS, UPDATE_USER_MUTATION } from "../Profile";

/*
test("render", async () => {
    const { asFragment } = render(<Profile />,
        {
            mocks: [
                {
                    request: {
                        query: QUERY_USER_DETAILS,
                        variables: {}
                    },
                    result: {
                        data: {
                            me: data.user[0]
                        }
                    }
                },
            ],
            initialEntries: ['/test']
        }
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTitle('loading')).not.toBeNull();
    // await waitForElementToBeRemoved(screen.queryByTitle('loading'));
});
*/

test("profile data shows up", async () => {
  const expectedUser = data.users.filter((i) => i.calendly_link)[0];
  const { asFragment } = render(<Profile />, {
    mocks: [
      {
        request: {
          query: QUERY_USER_DETAILS,
          variables: {},
        },
        result: {
          data: {
            me: expectedUser,
          },
        },
      },
    ],
    initialEntries: ["/test"],
  });
  await waitForElementToBeRemoved(screen.queryByTitle("loading"));
  expect(asFragment()).toMatchSnapshot();
  expect(screen.getByDisplayValue(expectedUser.email)).toBeInTheDocument();
  expect(screen.getByDisplayValue(expectedUser.name)).toBeInTheDocument();
  expectedUser.regions.forEach((zipcode, i) =>
    expect(screen.getByDisplayValue(zipcode)).toBeInTheDocument()
  );
});

test("change name", async () => {
  const expectedUser = data.users[0];
  const { asFragment } = render(<Profile />, {
    mocks: [
      {
        request: {
          query: QUERY_USER_DETAILS,
          variables: {},
        },
        result: {
          data: {
            me: expectedUser,
          },
        },
      },
      {
        request: {
          query: UPDATE_USER_MUTATION,
          variables: {
            email: expectedUser.email,
            name: "test_name",
            is_scout: expectedUser.is_scout,
            created_on: expectedUser.created_on,
            last_login: expectedUser.last_login,
            regions: expectedUser.regions,
            calendly_link: expectedUser.calendly_link,
            tours: [
              {
                tour_id: 2,
                tour_review_text:
                  "Thanks for the excellent tour. I'll rent ASAP",
                tour_review_stars: 5,
              },
            ],
            image: "user.png",
          },
        },
        result: {
          data: {
            updateUser: { ...expectedUser, name: "test_name" },
          },
        },
      },
    ],
    initialEntries: ["/test"],
  });
  await waitForElementToBeRemoved(screen.queryByTitle("loading"));
  window.alert = () => {};

  await act(async () => {
    const name = screen.getByDisplayValue(expectedUser.name);
    await userEvent.clear(name);
    await userEvent.type(name, "test_name");
    await screen.findByDisplayValue("test_name");
    await userEvent.click(screen.getByText("Save"));
  });

  await screen.findByText("Saving...");
  await screen.findByText("Save");

  expect(screen.getByDisplayValue("test_name")).toBeInTheDocument();
});

test("change calendly_link", async () => {
  const expectedUser = data.users[0];
  const { asFragment } = render(<Profile />, {
    mocks: [
      {
        request: {
          query: QUERY_USER_DETAILS,
          variables: {},
        },
        result: {
          data: {
            me: expectedUser,
          },
        },
      },
      {
        request: {
          query: UPDATE_USER_MUTATION,
          variables: {
            email: expectedUser.email,
            name: expectedUser.name,
            is_scout: expectedUser.is_scout,
            created_on: expectedUser.created_on,
            last_login: expectedUser.last_login,
            regions: expectedUser.regions,
            calendly_link: "new_calendly_link",
            tours: [
              {
                tour_id: 2,
                tour_review_text:
                  "Thanks for the excellent tour. I'll rent ASAP",
                tour_review_stars: 5,
              },
            ],
            image: "user.png",
          },
        },
        result: {
          data: {
            updateUser: { ...expectedUser, calendly_link: "new_calendly_link" },
          },
        },
      },
    ],
    initialEntries: ["/test"],
  });
  await waitForElementToBeRemoved(screen.queryByTitle("loading"));
  window.alert = () => {};

  await act(async () => {
    const calendly = screen.getByPlaceholderText(/paste calendly link here/i);
    await userEvent.type(calendly, "new_calendly_link");
    await screen.findByDisplayValue("new_calendly_link");
    await userEvent.click(screen.getByText("Save"));
  });

  await screen.findByText("Saving...");
  await screen.findByText("Save");

  expect(screen.getByDisplayValue("new_calendly_link")).toBeInTheDocument();
});

test("profile regions change", async () => {
  const expectedUser = data.users[0];
  const updatedRegions = expectedUser.regions.slice();
  updatedRegions[1] = 91112;
  const serverReply = JSON.parse(JSON.stringify(expectedUser));
  serverReply.regions = updatedRegions;
  const { asFragment } = render(<Profile />, {
    mocks: [
      {
        request: {
          query: QUERY_USER_DETAILS,
          variables: {},
        },
        result: {
          data: {
            me: expectedUser,
          },
        },
      },
      {
        request: {
          query: UPDATE_USER_MUTATION,
          variables: {
            email: expectedUser.email,
            name: expectedUser.name,
            is_scout: expectedUser.is_scout,
            created_on: expectedUser.created_on,
            last_login: expectedUser.last_login,
            regions: updatedRegions,
            calendly_link: expectedUser.calendly_link,
            tours: [
              {
                tour_id: 2,
                tour_review_text:
                  "Thanks for the excellent tour. I'll rent ASAP",
                tour_review_stars: 5,
              },
            ],
            image: "user.png",
          },
        },
        result: {
          data: {
            updateUser: serverReply,
          },
        },
      },
    ],
    initialEntries: ["/test"],
  });
  await waitForElementToBeRemoved(screen.queryByTitle("loading"));
  expect(asFragment()).toMatchSnapshot();

  expectedUser.regions.forEach((zipcode, i) =>
    expect(screen.getByDisplayValue(zipcode)).toBeInTheDocument()
  );
  window.alert = () => {};

  await act(async () => {
    let zipCodeElement = screen.getByDisplayValue(expectedUser.regions[1]);
    await userEvent.clear(zipCodeElement);
    await userEvent.type(screen.getByDisplayValue(0), "91112");
  });

  await screen.findByDisplayValue("91112");
  act(() => userEvent.click(screen.getByText("Save")));
  await screen.findByText("Saving...");
  await screen.findByText("Save");
  serverReply.regions.forEach((zipcode, i) =>
    expect(screen.getByDisplayValue(zipcode)).toBeInTheDocument()
  );
});

test("profile regions delete", async () => {
  const expectedUser = data.users[0];
  const updatedRegions = expectedUser.regions.slice(1);
  const serverReply = JSON.parse(JSON.stringify(expectedUser));
  serverReply.regions = updatedRegions;
  const { asFragment } = render(<Profile />, {
    mocks: [
      {
        request: {
          query: QUERY_USER_DETAILS,
          variables: {},
        },
        result: {
          data: {
            me: expectedUser,
          },
        },
      },
      {
        request: {
          query: UPDATE_USER_MUTATION,
          variables: {
            email: expectedUser.email,
            name: expectedUser.name,
            is_scout: expectedUser.is_scout,
            created_on: expectedUser.created_on,
            last_login: expectedUser.last_login,
            regions: expectedUser.regions.slice(1),
            calendly_link: expectedUser.calendly_link,
            tours: [
              {
                tour_id: 2,
                tour_review_text:
                  "Thanks for the excellent tour. I'll rent ASAP",
                tour_review_stars: 5,
              },
            ],
            image: "user.png",
          },
        },
        result: {
          data: {
            updateUser: serverReply,
          },
        },
      },
    ],
    initialEntries: ["/test"],
  });
  await waitForElementToBeRemoved(screen.queryByTitle("loading"));
  expect(asFragment()).toMatchSnapshot();

  expectedUser.regions.forEach((zipcode, i) =>
    expect(screen.getByDisplayValue(zipcode)).toBeInTheDocument()
  );
  window.alert = () => {};
  act(() => {
    userEvent.click(screen.getAllByRole("button", { name: /x/i })[0]);
  });

  await waitFor(() => expect(screen.queryByText("92092")).toBeNull());
  act(() => userEvent.click(screen.getByRole("button", { name: /save/i })));

  await screen.findByText("Saving...");
  await screen.findByText("Save");

  serverReply.regions.forEach((zipcode, i) =>
    expect(screen.getByDisplayValue(zipcode)).toBeInTheDocument()
  );
});

test("profile regions add", async () => {
  const expectedUser = data.users[0];
  const updatedRegions = [...expectedUser.regions, 12345];
  const serverReply = JSON.parse(JSON.stringify(expectedUser));
  serverReply.regions = updatedRegions;
  const { asFragment } = render(<Profile />, {
    mocks: [
      {
        request: {
          query: QUERY_USER_DETAILS,
          variables: {},
        },
        result: {
          data: {
            me: expectedUser,
          },
        },
      },
      {
        request: {
          query: UPDATE_USER_MUTATION,
          variables: {
            email: expectedUser.email,
            name: expectedUser.name,
            is_scout: expectedUser.is_scout,
            created_on: expectedUser.created_on,
            last_login: expectedUser.last_login,
            regions: [...expectedUser.regions, 12345],
            calendly_link: expectedUser.calendly_link,
            tours: [
              {
                tour_id: 2,
                tour_review_text:
                  "Thanks for the excellent tour. I'll rent ASAP",
                tour_review_stars: 5,
              },
            ],
            image: "user.png",
          },
        },
        result: {
          data: {
            updateUser: serverReply,
          },
        },
      },
    ],
    initialEntries: ["/test"],
  });
  await waitForElementToBeRemoved(screen.queryByTitle("loading"));
  expect(asFragment()).toMatchSnapshot();

  expectedUser.regions.forEach((zipcode, i) =>
    expect(screen.getByDisplayValue(zipcode)).toBeInTheDocument()
  );
  window.alert = () => {};
  await act(async () => {
    await userEvent.click(screen.getByPlaceholderText("add new region"));
    await userEvent.keyboard("12345");
  });

  await screen.findByDisplayValue("12345");
  act(() => userEvent.click(screen.getByRole("button", { name: /save/i })));

  await screen.findByText("Saving...");
  await screen.findByText("Save");

  serverReply.regions.forEach((zipcode, i) =>
    expect(screen.getByDisplayValue(zipcode)).toBeInTheDocument()
  );
});
