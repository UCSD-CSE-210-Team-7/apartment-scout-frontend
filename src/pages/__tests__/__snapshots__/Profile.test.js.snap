// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile data shows up 1`] = `
<DocumentFragment>
  <div
    style="display: flex;"
  >
    <div
      style="display: flex; flex-direction: column; padding: 0px 6em; align-items: center;"
    >
      <h1
        style="font-size: 3rem; margin: 1em 0em; padding: 1em 3em; color: rgb(2, 62, 138); background: rgb(173, 232, 244);"
      >
        My Profile
      </h1>
      <img
        alt="user"
        src="user.png"
        style="margin: 2em 0px; height: 20em; width: 20em;"
      />
      <label>
        <input
          style="display: none;"
          type="file"
        />
        <h1
          style="color: white; cursor: pointer;"
        >
          Edit profile photo
        </h1>
      </label>
      <div
        style="padding: 0.4em 0em; width: 10em; background: green; margin: 2em 0px 0px; color: white; font-size: 2em; text-align: center;"
      >
        Scout
      </div>
    </div>
    <div
      style="border: 1px solid black; display: flex; flex-direction: column; flex-grow: 1; padding: 2em; background: white; font-size: 2em;"
    >
      <h1>
        Personal Info
      </h1>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          name
        </label>
        <input
          type="text"
          value="Shubham Kulkarni"
        />
      </div>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          email
        </label>
        <input
          disabled=""
          readonly=""
          type="text"
          value="skulkarn@ucsd.edu"
        />
      </div>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          Calendly link
        </label>
        <input
          placeholder="Paste Calendly link here"
          style="width: 100%;"
          type="text"
          value="https://calendly.com/shubhamkulkarni01/60min"
        />
      </div>
      <div
        style="padding: 1em; display: flex; flex-direction: column;"
      >
        <p
          style="margin: 0px;"
        >
          regions
        </p>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="92092"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="60605"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="10010"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="2"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            placeholder="add new region"
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value=""
          />
        </div>
      </div>
      <button
        style="align-self: center; padding: 0.5em 1em; background: rgb(2, 62, 138); color: white; border-radius: 1em;"
      >
        Save
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`profile regions add 1`] = `
<DocumentFragment>
  <div
    style="display: flex;"
  >
    <div
      style="display: flex; flex-direction: column; padding: 0px 6em; align-items: center;"
    >
      <h1
        style="font-size: 3rem; margin: 1em 0em; padding: 1em 3em; color: rgb(2, 62, 138); background: rgb(173, 232, 244);"
      >
        My Profile
      </h1>
      <img
        alt="user"
        src="user.png"
        style="margin: 2em 0px; height: 20em; width: 20em;"
      />
      <label>
        <input
          style="display: none;"
          type="file"
        />
        <h1
          style="color: white; cursor: pointer;"
        >
          Edit profile photo
        </h1>
      </label>
      <div
        style="padding: 0.4em 0em; width: 10em; background: green; margin: 2em 0px 0px; color: white; font-size: 2em; text-align: center;"
      >
        Scout
      </div>
    </div>
    <div
      style="border: 1px solid black; display: flex; flex-direction: column; flex-grow: 1; padding: 2em; background: white; font-size: 2em;"
    >
      <h1>
        Personal Info
      </h1>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          name
        </label>
        <input
          type="text"
          value="Ajinkya Bokade"
        />
      </div>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          email
        </label>
        <input
          disabled=""
          readonly=""
          type="text"
          value="abokade@ucsd.edu"
        />
      </div>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          Calendly link
        </label>
        <input
          placeholder="Paste Calendly link here"
          style="width: 100%;"
          type="text"
          value=""
        />
      </div>
      <div
        style="padding: 1em; display: flex; flex-direction: column;"
      >
        <p
          style="margin: 0px;"
        >
          regions
        </p>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="92092"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="60605"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            placeholder="add new region"
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value=""
          />
        </div>
      </div>
      <button
        style="align-self: center; padding: 0.5em 1em; background: rgb(2, 62, 138); color: white; border-radius: 1em;"
      >
        Save
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`profile regions change 1`] = `
<DocumentFragment>
  <div
    style="display: flex;"
  >
    <div
      style="display: flex; flex-direction: column; padding: 0px 6em; align-items: center;"
    >
      <h1
        style="font-size: 3rem; margin: 1em 0em; padding: 1em 3em; color: rgb(2, 62, 138); background: rgb(173, 232, 244);"
      >
        My Profile
      </h1>
      <img
        alt="user"
        src="user.png"
        style="margin: 2em 0px; height: 20em; width: 20em;"
      />
      <label>
        <input
          style="display: none;"
          type="file"
        />
        <h1
          style="color: white; cursor: pointer;"
        >
          Edit profile photo
        </h1>
      </label>
      <div
        style="padding: 0.4em 0em; width: 10em; background: green; margin: 2em 0px 0px; color: white; font-size: 2em; text-align: center;"
      >
        Scout
      </div>
    </div>
    <div
      style="border: 1px solid black; display: flex; flex-direction: column; flex-grow: 1; padding: 2em; background: white; font-size: 2em;"
    >
      <h1>
        Personal Info
      </h1>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          name
        </label>
        <input
          type="text"
          value="Ajinkya Bokade"
        />
      </div>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          email
        </label>
        <input
          disabled=""
          readonly=""
          type="text"
          value="abokade@ucsd.edu"
        />
      </div>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          Calendly link
        </label>
        <input
          placeholder="Paste Calendly link here"
          style="width: 100%;"
          type="text"
          value=""
        />
      </div>
      <div
        style="padding: 1em; display: flex; flex-direction: column;"
      >
        <p
          style="margin: 0px;"
        >
          regions
        </p>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="92092"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="60605"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            placeholder="add new region"
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value=""
          />
        </div>
      </div>
      <button
        style="align-self: center; padding: 0.5em 1em; background: rgb(2, 62, 138); color: white; border-radius: 1em;"
      >
        Save
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`profile regions delete 1`] = `
<DocumentFragment>
  <div
    style="display: flex;"
  >
    <div
      style="display: flex; flex-direction: column; padding: 0px 6em; align-items: center;"
    >
      <h1
        style="font-size: 3rem; margin: 1em 0em; padding: 1em 3em; color: rgb(2, 62, 138); background: rgb(173, 232, 244);"
      >
        My Profile
      </h1>
      <img
        alt="user"
        src="user.png"
        style="margin: 2em 0px; height: 20em; width: 20em;"
      />
      <label>
        <input
          style="display: none;"
          type="file"
        />
        <h1
          style="color: white; cursor: pointer;"
        >
          Edit profile photo
        </h1>
      </label>
      <div
        style="padding: 0.4em 0em; width: 10em; background: green; margin: 2em 0px 0px; color: white; font-size: 2em; text-align: center;"
      >
        Scout
      </div>
    </div>
    <div
      style="border: 1px solid black; display: flex; flex-direction: column; flex-grow: 1; padding: 2em; background: white; font-size: 2em;"
    >
      <h1>
        Personal Info
      </h1>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          name
        </label>
        <input
          type="text"
          value="Ajinkya Bokade"
        />
      </div>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          email
        </label>
        <input
          disabled=""
          readonly=""
          type="text"
          value="abokade@ucsd.edu"
        />
      </div>
      <div
        style="padding: 1em;"
      >
        <label
          style="margin: 0px;"
        >
          Calendly link
        </label>
        <input
          placeholder="Paste Calendly link here"
          style="width: 100%;"
          type="text"
          value=""
        />
      </div>
      <div
        style="padding: 1em; display: flex; flex-direction: column;"
      >
        <p
          style="margin: 0px;"
        >
          regions
        </p>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="92092"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value="60605"
          />
          <button
            style="color: white; background: red; border-radius: 0.5em; margin: 0em 0.5em; padding: 0em 0.5em;"
          >
            X
          </button>
        </div>
        <div
          style="padding: 0.5em;"
        >
          <input
            placeholder="add new region"
            style="text-align: center; border: 0px; border-bottom: 1px solid black;"
            type="number"
            value=""
          />
        </div>
      </div>
      <button
        style="align-self: center; padding: 0.5em 1em; background: rgb(2, 62, 138); color: white; border-radius: 1em;"
      >
        Save
      </button>
    </div>
  </div>
</DocumentFragment>
`;
